src/writer/update-proposal.ts

```diff
@@ -0,0 +1,199 @@
+import isEqual from 'lodash/isEqual';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { getAddress } from '@ethersproject/address';
+import { jsonParse } from '../helpers/utils';
+import db from '../helpers/mysql';
+import { getSpace, getProposal } from '../helpers/actions';
+import log from '../helpers/log';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import {
+  flaggedAddresses,
+  flaggedProposalTitleKeywords,
+  flaggedProposalBodyKeywords
+} from '../helpers/moderation';
+
+const scoreAPIUrl = process.env.SCORE_API_URL || 'https://score.snapshot.org';
+
+export function validateChoices({ type, choices }): boolean {
+  if (type && choices.length > 0) {
+    switch (type) {
+      case 'basic':
+        return isEqual(['For', 'Against', 'Abstain'], choices);
+      default:
+        return choices.length > 0;
+    }
+  } else {
+    return false;
+  }
+}
+
+// We don't need most of the checks used https://github.com/snapshot-labs/snapshot-sequencer/blob/89992b49c96fedbbbe33b42041c9cbe5a82449dd/src/writer/proposal.ts#L62
+// because we assume that those checks were already done during the proposal creation
+export function getSpaceUpdateError({ type, space }): string | undefined {
+  const { voting = {} } = space;
+
+  if (voting.type && type !== voting.type) return 'space voting type mismatch';
+
+  return undefined;
+}
+
+export function isScamDetected({ address, name, body }): boolean {
+  const proposalNameLC = name.toLowerCase();
+  const proposalBodyLC = body.toLowerCase();
+  const addressLC = address.toLowerCase();
+
+  return (
+    flaggedAddresses.includes(addressLC) ||
+    flaggedProposalBodyKeywords.some(keyword => proposalBodyLC.includes(keyword)) ||
+    flaggedProposalTitleKeywords.some(keyword => proposalNameLC.includes(keyword))
+  );
+}
+
+export function isAddressAuthorized(address: string, space: any): boolean {
+  const addressLC = address.toLowerCase();
+  const members = [
+    ...(space.members || []),
+    ...(space.admins || []),
+    ...(space.moderators || [])
+  ].map(member => member.toLowerCase());
+
+  return members.includes(addressLC);
+}
+
+export async function checkAuthorization(address: string, space: any): Promise<string | undefined> {
+  const onlyAuthors = space.filters?.onlyMembers;
+  const isAuthorized = isAddressAuthorized(address, space);
```

Comment:
We dont want this, only author of the proposal should be able to update proposal